base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ScriptGeneratorViewModel.java

```diff
@@ -284,24 +304,38 @@ public String getText(Object element) {
 		    }
 		};
 	}
+	
+	private PropertyChangeListener actionChangeListener = new PropertyChangeListener() {
```

Comment:
I think lambdas can be used to replace the usage of anonymous inner classes.

Reply:
The inner class is used because actionChangeListener must be removed everytime we rebind validity checks to make sure we are not binding it multiple times. Creating it in this way allows us to easily have the variable as a reference so we can remove and add each time. It is also the same handler for multiple different property change listeners so is nice to have just one reference to reduce repeated code.

Reply:
I was not clear enough. I meant something like this `private PropertyChangeListener actionChangeListener = evt->actionChangeHandler(
			this.viewTable, this.btnGetValidityErrors, this.btnGenerateScript);`

Reply:
Oh I see, my mistake, yes that makes the code simpler for sure. All changed now.